imageprocessor-api/src/test/java/pl/edu/uj/kimage/api/ApiTest.java

```diff
@@ -0,0 +1,39 @@
+package pl.edu.uj.kimage.api;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.util.Arrays;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class ApiTest {
+
+    @Test
+    public void shouldCorrectlySerializeApiClasses() throws JsonProcessingException {
+        //Given
+        StepDependency stepDependency = new StepDependency(0,Object.class);
+        Step step = new Step(0,"pluginName", Arrays.asList(stepDependency));
+        ProcessingSchedule processingSchedule = new ProcessingSchedule(Arrays.asList(step));
+        ObjectMapper objectMapper = new ObjectMapper();
+        //When
+        String valueAsString = objectMapper.writeValueAsString(processingSchedule);
+        //Then
+        assertThat(valueAsString).matches("^\\{.*\\}$");
+    }
+
+    @Test
+    public void shouldCorrectlyDeserializeApiClasses() throws IOException {
+        StepDependency stepDependency = new StepDependency(0,Object.class);
+        Step step = new Step(0,"pluginName", Arrays.asList(stepDependency));
+        ProcessingSchedule processingSchedule = new ProcessingSchedule(Arrays.asList(step));
+        ObjectMapper objectMapper = new ObjectMapper();
+        String valueAsString = objectMapper.writeValueAsString(processingSchedule);
+        //When
+        ProcessingSchedule readValue = objectMapper.readValue(valueAsString, ProcessingSchedule.class);
+        //Then
+        assertThat(readValue).isEqualTo(processingSchedule);
+    }
+}
```

Comment:
I understand that, but I don't see any connection with compatibility. You just add a conversion.